packages/amplify-migration-tests/src/__tests__/migration_tests_v10/scaffold.test.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  createNewProjectDir,
+  deleteProject,
+  deleteProjectDir,
+} from '@aws-amplify/amplify-e2e-core';
+import { versionCheck, allowedVersionsToMigrateFrom } from '../../migration-helpers';
+import { initJSProjectWithProfileV10 } from '../../migration-helpers-v10/init';
+
+describe('amplify migration test scaffold v10', () => {
+  let projRoot: string;
+
+  beforeAll(async () => {
+    const migrateFromVersion = { v: 'unintialized' };
+    const migrateToVersion = { v: 'unintialized' };
+    await versionCheck(process.cwd(), false, migrateFromVersion);
+    await versionCheck(process.cwd(), true, migrateToVersion);
+    expect(migrateFromVersion.v).not.toEqual(migrateToVersion.v);
+    expect(allowedVersionsToMigrateFrom).toContain(migrateFromVersion.v);
+  });
+
+  beforeEach(async () => {
+    projRoot = await createNewProjectDir('scaffoldTest');
+  });
+
+  afterEach(async () => {
+    await deleteProject(projRoot, null, true);
+    deleteProjectDir(projRoot);
+  });
+
+  it('...should init a project', async () => {
+    await initJSProjectWithProfileV10(projRoot, { name: 'scaffoldTest' });
```

Comment:
in v10, we have custom init flow that is different than in v4/v5/v6.. so this test showcases that